genestack_client/genestack_connection.py

```diff
@@ -221,7 +220,10 @@ def __init__(self, connection, application_id=None):
         if len(self.application_id.split('/')) != 2:
             raise GenestackException('Invalid application ID, expect "{vendor}/{application}" got: %s' % self.application_id)
 
-    def __invoke(self, path, post_data):
+    def __invoke(self, path, post_data, trace=None):
+        self.connection.opener.addheaders = [('gs-extendSession', 'true')]
+        if trace:
+            self.connection.opener.addheaders.append(('gs-trace', 'true'))
```

Comment:
We can rewrite it using `BaseHandler.protocol_open` method (see here: https://docs.python.org/2/library/urllib2.html#basehandler-objects).

But this seems like an overkill, so I propose just to move this logic into `connection.open` method, and send there just a multimap or another headers object implementation containing desired headers, so that the opener will be a private field (can we change the field declaration to be private?)